R/method-results.R

```diff
@@ -19,58 +20,41 @@ compileResults.OUTRIDER <- function(object, padjCutoff=0.05, zScoreCutoff=0,
         round <- 2
     }
     
-    if(isFALSE(all)){
-        abByGene <- aberrant(object, 
-                padjCutoff=padjCutoff, zScoreCutoff=zScoreCutoff, by="gene")
-        abBySample <- aberrant(object, 
-                padjCutoff=padjCutoff, zScoreCutoff=zScoreCutoff, by="sample")
-        object <- object[abByGene > 0, abBySample > 0]
-    }
-    
-    if(nrow(object) == 0){
-        if(isFALSE(all)){
-            warning('No significant events: use all=TRUE to print all events.')
-        } else {
-            warning('Please provide an object with at least one feature.')
-        }
-        return(data.table(
-                geneID=NA_character_,
-                sampleID=NA_character_,
-                pValue=NA_real_,
-                padjust=NA_real_,
-                zScore=NA_real_,
-                l2fc=NA_real_,
-                rawcounts=NA_integer_,
-                normcounts=NA_real_,
-                meanCorrected=NA_real_,
-                theta=NA_real_,
-                aberrant=NA,
-                AberrantBySample=NA_integer_,
-                AberrantByGene=NA_integer_,
-                padj_rank=NA_real_)[0])
+    # compute FDR on subset if subset is specified
+    if(!is.null(genesToTest)){
+        subsetName <- ifelse(is.null(subsetName), "subset", subsetName)
+        fdrSet <- subsetName
+        object <- padjOnSubset(ods=object, genesToTest=genesToTest, 
```

Comment:
Remove it from here and only put it into the OUTRIDER function as well as in the computePvalue function.